tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -76,7 +53,7 @@ static bool IsIntPtrCtor (MethodDefinition d)
 			return isSingle || isDouble;
 		}
 
-		bool IsNSObjectDerived (TypeReference? typeReference)
+		public static bool IsNSObjectDerived (TypeReference? typeReference, ConstructorTransforms? self = null)
```

Comment:
We could have passed the callable rather than the entire class, since now we have a quite strong connection between the ConstructorTransforms implementation and the method implementation. Or return a out param with the warning event args and decide the called what to do.

Reply:
Yes. I thought about a couple different ways of doing this and chose the path of least resistance. This has the unsavory bit of having the coupling of the two classes and the optional parameter. I considered this somewhat better than, say, making a duplicate of the function somewhere else but without the element - or passing in the event handler instead of the instance.

Reply:
You can do something like this:
```csharp
public static bool IsNSObjectDerived (TypeReference? typeReference, out ConstructorTransforms? WarningEventArg warning);
```

Then use the event args if you want/can